components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -327,4 +394,66 @@ public Response deleteRegistryEntry(String registryId, String entryId, MessageCo
                 " in :" + registryId + " by:" + user);
         return Response.ok().entity("Successfully deleted the endpoint registry entry").build();
     }
+
+    private boolean isValidEndpointDefinition(URL definitionURL, InputStream definitionFileInputStream,
+                                              String definitionType) {
+        String definitionContent = null;
+        try {
+            if (definitionURL != null) {
+                definitionContent = IOUtils.toString(definitionURL.openStream());
+            } else if (definitionFileInputStream != null) {
```

Comment:
shouldn't we change the order here?

Reply:
The order does not matter here. It is handled before calling this validation method.

Reply:
by looking at this method only, one would get the impression that the URL is given priority. So shall we.change the order with a comment?

Reply:
yeah. no harm in changing that. I ll will

Reply:
updated